### DISCUSSION TITLE
Runtimes for the RPi0

### DISCUSSION THREAD

**Discussion by docdoob** [AdaCore/bb-runtimes#50] [Ideas]
M. Fabien-Chouteau,
I've used your repository to create runtimes for the Raspberry Pi Zero (RPi0).  It is in https://github.com/docdoob/Ada-on-RPi0.  I took your RPi2 runtime build and modified it, deleting some features.  

Are you interested?

I have tested the Zero Footprint profile, but neither Ravenscar runtime is tested.

Regards,
John Dubery

**Comment by Fabien-Chouteau** [comment]
Hello @docdoob,

The RPi0 is not necessarily a target that we are looking for in this repository.
The reason is that this board is really meant to run Linux and if you want to program it in Ada, the best way is to do it via the raspbian GNAT packages.

**Reply by docdoob** [reply]
M. Fabien-Chouteau,
I was surprised that you are not interested in the RPi0 as a target.  It seems a good basis for medium-sized "gadgets" and surely it is not powerful enough to be a development machine.   Are you interested in the Pico?
I've been using the cross-compiler on Linux 63b but I have now installed GNAT on a RPi3.  The rules for project files must be a bit different as I can't link automatically from gprbuild.  But the code generated for a package is very similar, which is comforting!

**Reply by Fabien-Chouteau** [reply]
> I was surprised that you are not interested in the RPi0 as a target. It seems a good basis for medium-sized "gadgets" and surely it is not powerful enough to be a development machine.

It is not that we are not interested at all, but as I said the Raspberries are Linux machines.

> Are you interested in the Pico?

Yes of course, this micro-controller is much more inline with the run-times we have here. In fact the Ada community aleady has a good support for the Pico:
 - https://alire.ada.dev/crates/pico_examples
 - https://github.com/AdaCore/bb-runtimes/pull/51

> I've been using the cross-compiler on Linux 63b but I have now installed GNAT on a RPi3. The rules for project files must be a bit different as I can't link automatically from gprbuild. But the code generated for a package is very similar, which is comforting!

It should work out of the box. Maybe a issue to report to gprbuild or the package maintainers.
